.github/workflows/_build.yaml

```diff
@@ -0,0 +1,134 @@
+# Note: if the build workflow needs to access secrets, they need to be passed by the caller using
+# `secrets: inherit`. See also
+#
+# https://docs.github.com/en/actions/using-workflows/reusing-workflows
+# https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions
+#
+# for the security recommendations.
+
+name: Build the package
+on:
+  workflow_call:
+    outputs:
+      artifacts-sha256:
+        description: The hash of the artifacts
+        value: ${{ jobs.build.outputs.artifacts-sha256 }}
+permissions:
+  contents: read
+env:
+  ARTIFACT_OS: ubuntu-latest # The default OS for release.
+  ARTIFACT_PYTHON: '3.11' # The default Python version for release.
+
+jobs:
+  build:
+    outputs:
+      artifacts-sha256: ${{ steps.compute-hash.outputs.artifacts-sha256 }}
+    name: Build Python ${{ matrix.python }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        # It is recommended to pin a Runner version specifically:
+        # https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
+        os: [ubuntu-latest]
+        python: ['3.11']
+    steps:
+
+    - name: Check out repository
+      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@13ae5bb136fac2878aff31522b9efb785519f984 # v4.3.0
+      with:
+        python-version: ${{ matrix.python }}
+
+    # Using the Makefile assumes an activated virtual environment, which doesn't exist
+    # when running in an Action environment (https://github.com/actions/setup-python/issues/359).
+    # Instead we create an empty .venv folder so that the Makefile continues to function
+    # while Python operates within the runner's global environment. It is safe to ignore
+    # warnings from the Makefile about the missing virtual environment.
+    - name: Create empty virtual environment for Actions
+      run: mkdir .venv
+    - name: Install dependencies
+      run: make setup
+
+    # Audit all currently installed packages for security vulnerabilities.
+    - name: Audit installed packages
+      run: make audit
+
+    # Build the sdist and wheel distribution of the package and docs as a zip file.
+    # We don't need to check and test the package separately because `make dist` runs
+    # those targets first and only builds the package if they succeed.
+    - name: Build the package
+      run: make dist
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+
+    # Generate the requirements.txt that contains the hash digests of the dependencies and
+    # generate the SBOM using CyclonDX SBOM generator.
+    - name: Generate requirements.txt and SBOM
+      if: matrix.os == env.ARTIFACT_OS && matrix.python == env.ARTIFACT_PYTHON
+      run: make requirements sbom
+
+    # Remove the old requirements.txt file (which includes _all_ packages) and generate a
```

Comment:
Should we only perform the removal before generating the `requirements.txt` and the SBOM in the above step?

Reply:
> Should we only perform the removal before generating the `requirements.txt` and the SBOM in the above step?

I think I misunderstand while looking at this step. I believe the `requirements.txt` in the `Generate requirements.txt and SBOM` step will contain the packages that are not runtime dependencies for Macaron. It is different from the `requirements.txt` file used in the `Prune packages and generate required requirements.txt` step. I will resolve the conversation if you can confirm that I understand it correctly 👍

Reply:
>I believe the `requirements.txt` in the `Generate requirements.txt and SBOM` step will contain the packages that are not runtime dependencies for Macaron. It is different from the `requirements.txt` file used in the `Prune packages and generate required requirements.txt`

Yes, that's correct.